qa/rpc-tests/paymentdisclosure.py

```diff
@@ -0,0 +1,233 @@
+        recipients = [{"address":myzaddr, "amount":Decimal('40.0')-Decimal('0.0001')}]
+        myopid = self.nodes[0].z_sendmany(mytaddr, recipients)
+        txid = self.wait_and_assert_operationid_status(0, myopid)
+
+        # Check the tx has joinsplits
+        assert( len(self.nodes[0].getrawtransaction("" + txid, 1)["vjoinsplit"]) > 0 )
+
+        # Sync mempools
+        self.sync_all()
+
+        # Confirm that you can't create a payment disclosure for an unconfirmed tx
+        try:
+            self.nodes[0].z_getpaymentdisclosure(txid, 0, 0)
+            assert(False)
+        except JSONRPCException as e:
+            errorString = e.error['message']
+            assert_equal("Transaction has not been confirmed yet" in errorString, True)
+
+        try:
+            self.nodes[1].z_getpaymentdisclosure(txid, 0, 0)
+            assert(False)
+        except JSONRPCException as e:
+            errorString = e.error['message']
+            assert_equal("Transaction has not been confirmed yet" in errorString, True)
+
+        # Mine tx
+        self.nodes[0].generate(1)
+        self.sync_all()
+
+        # Confirm that Node 1 cannot create a payment disclosure for a transaction which does not impact its wallet
+        try:
+            self.nodes[1].z_getpaymentdisclosure(txid, 0, 0)
+            assert(False)
+        except JSONRPCException as e:
+            errorString = e.error['message']
+            assert_equal("Transaction does not belong to the wallet" in errorString, True)
+
+        # Check that an invalid joinsplit index is rejected
+        try:
+            self.nodes[0].z_getpaymentdisclosure(txid, 1, 0)
+            assert(False)
+        except JSONRPCException as e:
+            errorString = e.error['message']
+            assert_equal("Invalid js_index" in errorString, True)
+
+        try:
+            self.nodes[0].z_getpaymentdisclosure(txid, -1, 0)
+            assert(False)
+        except JSONRPCException as e:
+            errorString = e.error['message']
+            assert_equal("Invalid js_index" in errorString, True)
+
+        # Check that an invalid output index is rejected
+        try:
+            self.nodes[0].z_getpaymentdisclosure(txid, 0, 2)
+            assert(False)
+        except JSONRPCException as e:
+            errorString = e.error['message']
+            assert_equal("Invalid output_index" in errorString, True)
+
+        try:
+            self.nodes[0].z_getpaymentdisclosure(txid, 0, -1)
+            assert(False)
+        except JSONRPCException as e:
+            errorString = e.error['message']
+            assert_equal("Invalid output_index" in errorString, True)
+
+        # Ask Node 0 to create and validate a payment disclosure for output 0
+        message = "Here is proof of my payment!"
+        pd = self.nodes[0].z_getpaymentdisclosure(txid, 0, 0, message)
+        result = self.nodes[0].z_validatepaymentdisclosure(pd)
+        assert( result["valid"] == True )
```

Comment:
Use ``assert_equal`` here and below in this test.

Reply:
ok